src/ConnectedRouter.js

```diff
@@ -35,7 +35,7 @@ const createConnectedRouter = (structure) => {
         } = props.history.location
 
         // If we do time travelling, the location in store is changed but location in history is not changed
-        if (pathnameInHistory !== pathnameInStore || searchInHistory !== searchInStore || hashInHistory !== hashInStore) {
+        if (props.history.action === 'PUSH' && (pathnameInHistory !== pathnameInStore || searchInHistory !== searchInStore || hashInHistory !== hashInStore)) {
```

Comment:
Would be nice to have here some detailed comment to explain why it is important to check the action.

Reply:
Could we test it to be sure it wont break the next time?